MetaMask/metamask-extension
# refactor: Send signature metrics as event fragments in middleware & feat: Add signature alert metrics to events

## **Description**

- If the RPC request is a signature, send the event using event fragments so that we can add additional props before sending the final event 
  - Pass MetaMetricsController to createRPCMethodTrackingMiddleware
- Add signature alert metrics to events

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26597?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2718
Blocked by: https://github.com/MetaMask/core/pull/4636/files and following PRs to update packages 

## **Manual testing steps**

1. Turn on metametrics through MM settings
2. Go to test-dapp
3. Trigger signature with alerts (e.g. Sign-in with Ethereum (Bad Domain))
4. Observe new alert metrics in Segment/Mixpanel Signature Rejected/Approved events

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
